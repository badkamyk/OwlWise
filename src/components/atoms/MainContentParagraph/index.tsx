import classnames from "classnames";
import { ContentTypographyType } from "../ContentHeading/ContentTypographyType";

export default function MainContentParagraph({
	text,
	className = "mb-4",
	size = "md",
	color = "gray",
}: ContentTypographyType) {
	return (
		<p
			data-testid="main-content-paragraph"
			className={classnames(className, {
				"text-gray-900": color === "gray",
				"text-blue-700": color === "blue",
				"text-white": color === "white",
				"text-sm": size === "sm",
				"text-md": size === "md",
				"text-lg": size === "lg",
				"text-xl": size === "xl",
			})}
		>
			{text}
		</p>
	);
}
