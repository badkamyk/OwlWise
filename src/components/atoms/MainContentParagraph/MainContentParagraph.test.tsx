import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import MainContentParagraph from "./index";

describe("MainContentParagraph", () => {
	it("MainContentParagraph with only text prop", () => {
		const { container } = render(<MainContentParagraph text={"Hello World"} />);
		expect(screen.getByText("Hello World")).toBeTruthy();
		expect(
			container.getElementsByClassName("mb-4 text-md text-gray-900")
		).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	it("MainContentParagraph with props", () => {
		const { container } = render(
			<MainContentParagraph
				text="Hello World"
				color={"blue"}
				size={"lg"}
				className={"mb-8"}
			/>
		);
		expect(
			container.getElementsByClassName("mb-8 text-lg text-blue-700")
		).toBeTruthy();
		expect(screen.getByText("Hello World")).toBeTruthy();
		expect(container).toMatchSnapshot();
	});
});
