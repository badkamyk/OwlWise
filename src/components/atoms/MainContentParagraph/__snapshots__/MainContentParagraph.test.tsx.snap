// Vitest Snapshot v1

exports[`MainContentParagraph > MainContentParagraph with only text prop 1`] = `
<div>
  <p
    class="mb-4 text-gray-900 text-md"
  >
    Hello World
  </p>
</div>
`;

exports[`MainContentParagraph > MainContentParagraph with props 1`] = `
<div>
  <p
    class="mb-8 text-blue-700 text-lg"
  >
    Hello World
  </p>
</div>
`;
